dht.go

```diff
@@ -498,6 +522,33 @@ func (dht *IpfsDHT) SetMode(m DHTMode) error {
 	}
 }
 
+func (dht *IpfsDHT) dynamicModeSwitching(ctx context.Context) goprocess.Process {
+	proc := goprocessctx.WithContext(ctx)
+	watch := func(proc goprocess.Process) {
+		for {
+			select {
+			case ev := <-dht.subscriptions.evtLocalRoutability.Out():
+				var err error
+				switch ev.(type) {
+				case event.EvtLocalRoutabilityPrivate, event.EvtLocalRoutabilityUnknown:
```

Comment:
I'm a little nervous about flapping here. Could we add a check that this hasnt happened too recently, and throw an error log if it does?

Reply:
We could also debounce this:

1. Set a local "routable" variable each time.
2. Reset a timeout each time we receive an event.
3. Only change modes when we haven't received an event for N seconds.

Reply:
The debouncing sounds sensible. What do we want the value of N to be? 1 minute?

Reply:
Implemented.